INSTALL

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
+#
+# $COPYRIGHT$
+# $HEADER$
+#
+
+
+For More Information
+====================
+
+This file is a *very* short overview of building and installing UCX
+and building MPI/SHMEM libraries with it.  Much more information is
+available on the UCX web site and github wiki
+
+    http://openucx.org/
+
+    -and-
+
+    https://github.com/openucx/ucx/wiki
+
+
+Developer Builds
+================
+
+shell$ ./autogen.sh
+shell$ ./contrib/configure-devel --prefix=$PWD/install-debug
+
+You will need very recent versions of GNU Autoconf, Automake, and
+Libtool. 
+
+*** NOTE: Developer's copies of UCX typically include a large
+performance penalty at run-time because of extra debugging overhead.
+
+
+User Builds
+===========
+
+Building UCX is typically a combination of running "configure"
+and "make".  Execute the following commands to install the UCX
+system from within the directory at the top of the tree:
+
+shell$ ./autogen.sh
+shell$ ./contrib/configure-release --prefix=/where/to/install
+shell$ make all install
+
+If you need special access to install, then you can execute "make
+all" as a user with write permissions in the build tree, and a
+separate "make install" as a user with write permissions to the
+install tree.
+
+Compiling support for various networks or other specific hardware may
+require additional command line flags when running configure.
+
+Parallel builds are also supported (although some versions of "make",
+such as GNU make, will only use the first target listed on the command
+line when executable parallel builds).  For example (assume GNU make):
+
+shell$ make -j 4 all
+shell$ make install
+
+Parallel make is generally only helpful in the build phase (i.e.,
+"make all"); the installation process (i.e., "make install") is mostly
+serial and does not benefit much from parallelization.
+
+# Build rpm package
+shell$ contrib/buildrpm.sh -s -b
+ 
+# Build deb package
+shell$ dpkg-buildpackage -us -uc
+ 
+# Build Doxygen documentation
+shell$ make docs
+
+
+Compiling OpenMPI with UCX library
+==================================
+
+Open MPI package includes two public software layers: MPI and OpenSHMEM.
+The UCX support in OpenMPI is not upstream yet, it is work in progress.
+Current work on OpenMPI/UCX integration is going for OpenSHMEM API and can be found here:
```

Comment:
Current work on OpenMPI/UCX integration is ongoing for the OpenSHMEM API and can be found here: